src/net_processing.cpp

```diff
@@ -4342,8 +4372,17 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                     blockTxnMsg << txn;
                     fProcessBLOCKTXN = true;
                 } else {
-                    req.blockhash = pindex->GetBlockHash();
-                    m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::GETBLOCKTXN, req));
+                    // We will try to round-trip any compact blocks we get on failure,
+                    // as long as it's first, or not too large and as long as we chose
+                    // the peer for high-bandwidth relay.
+                    if (first_in_flight ||
+                        (pfrom.m_bip152_highbandwidth_to && req.indexes.size() <= MAX_GETBLOCKTXN_TXN_AFTER_FIRST_IN_FLIGHT)) {
+                        req.blockhash = pindex->GetBlockHash();
+                        m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::GETBLOCKTXN, req));
```

Comment:
This could be a `else if` of the outer if statement. That would avoid the nesting here.

Reply:
done